src/mlpack/core/data/pre_processing_impl.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef MLPACK_CORE_DATA_DATA_PRE_PROCESSING_IMPL_HPP
+#define MLPACK_CORE_DATA_DATA_PRE_PROCESSING_IMPL_HPP
+
+#include "pre_processing.hpp"
+
+namespace mlpack {
+namespace data {
+  Data::Data(std::string file_path)
+  {
+    // Constructor
+    path = file_path;
+    mlpack::data::Load(path, data, true);
+    data = data.submat(0, 1, data.n_rows - 1, data.n_cols - 1);
+    getHeaders();
+    getNumericAttributes();
+  }
+
+  void Data:: info()
+  {
+    // Returns Index range, list of headers
+    std::cout << typeid(data).name() << std::endl;
```

Comment:
This isn't guaranteed to be a readable description of the name---it might be tedious, but we should probably find another solution here.